### PR TITLE
Update celo-bls-go version to v0.3.4

### PR DESCRIPTION
**Pull request by mstraka100**
### Description

Updates celo-bls-go dependency to v0.3.4, which is identical to v0.3.4 except it is in sync with the commit in https://proxy.golang.org/github.com/celo-org/celo-bls-go/@v/v0.3.4.info. This is to fix https://github.com/celo-org/celo-blockchain/issues/2005.

### Tested

No functional code changes were made. CI tests should be sufficient.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -108,18 +108,32 @@ github.com/buraksezer/consistent v0.0.0-20191006190839-693edf70fd72/go.mod h1:OE
 github.com/c-bata/go-prompt v0.2.2/go.mod h1:VzqtzE2ksDBcdln8G7mk2RX9QyGjH+OVqOCSiVIqS34=
 github.com/celo-org/celo-bls-go v0.3.3 h1:kOkm/BeM0YwHWUTwSLprxhk1IFq/ZjaWVtW6jTZ0+ys=
 github.com/celo-org/celo-bls-go v0.3.3/go.mod h1:eoMAORYWgZ5HOo3Z0bIAv/nbPtj/eArIO0nh7XFx7rk=
+github.com/celo-org/celo-bls-go v0.3.4 h1:slNePT/gVjgUi7f8M4KTwBz/YYgv3JWU6XqyY0xKN84=
```
Comment (carterqw2) [inline go.sum:111]: Hmm, that's weird that we have dependencies for two different versions for all `celo-bls-go-*` packages now.

**Comment by palango** [comment]
Did you create a new `celo-bls-go` release for this or did that already exist?

**Comment by mstraka100** [comment]
> Did you create a new `celo-bls-go` release for this or did that already exist?

I made a new version (v0.3.4) which was functionally identical to the last (v0.3.3). The celo-bls-go subrepos are still at v0.3.3 since they didn't need to be changed and updating them would just have introduced unnecessary complexity

**Comment by palango** [comment]
> > Did you create a new `celo-bls-go` release for this or did that already exist?
> 
> I made a new version (v0.3.4) which was functionally identical to the last (v0.3.3). The celo-bls-go subrepos are still at v0.3.3 since they didn't need to be changed and updating them would just have introduced unnecessary complexity

Will that tag be merged into master as well?

**Review by palango — APPROVED** [review]
👍

**Comment by mstraka100** [comment]
> > > > Did you create a new `celo-bls-go` release for this or did that already exist?
> > > 
> > > 
> > > I made a new version (v0.3.4) which was functionally identical to the last (v0.3.3). The celo-bls-go subrepos are still at v0.3.3 since they didn't need to be changed and updating them would just have introduced unnecessary complexity
> > 
> > 
> > Will that tag be merged into master as well?
> 
> What do you mean by this? The current PR just updates the `celo-bls-go` dependency to v0.3.4, keeping the other `celo-bls-go-*` repos as v0.3.3.

If you're asking will it be merged into master of celo-bls-go, that's a good question. The branch of v0.3.* is not intended to be merged into master. It is a stopgap solution to avoid using unaudited cryptography code (BLS batch verification) as a dependency in celo-blockchain, which is in master on celo-bls-go as v0.6.*. It may be worth discussing whether we want to simply use this version in celo-blockchain anyways to point to master of celo-bls-go since it might be a while before we get this code audited with its deprioritization. We could just make sure to be explicit in the celo-bls-go repo description that this segment is unaudited.

**Pull request merged**